src/test/java/org/opensearch/knn/index/VectorDataTypeIT.java

```diff
@@ -425,6 +425,51 @@ public void testKNNScriptScoreWithInvalidByteQueryVector() throws Exception {
         );
     }
 
+    @SneakyThrows
```

Comment:
I would prefer to keep the IT tests to be minimal by not including error cases.
If the test can be covered in unit test `KNNQueryBuilderTests`, can we remove tests in `VectorDataTypeIT`?

Reply:
I would say its better have the IT and unit tests. Both have their use. So, I would like to keep both of them.

Reply:
I'd like to keep the IT cases here, it's still a potential behavior from customers, it's good to have the IT to verify from end to end.

Reply:
What is the case where unit test fails to catch but IT can catch here?
If the issue can be caught using simple unit test, why do we want to add expensive/duplicated IT test?

Reply:
@heemin32 The unit test cannot verify the return response code which is expected to be 400(BAD_REQUEST), It can be safer to make sure the right response code is returned to end user by IT in k-NN plugin level. Ref: https://github.com/opensearch-project/OpenSearch/blob/2de44a7b771c9b8b59f57069d0fdfdf9ee818ec2/libs/core/src/main/java/org/opensearch/ExceptionsHelper.java#L99-L100

Reply:
The responsibility of return code is not in your code but in the OpenSearch framework. All you need to do is throwing correct exception.

Reply:
@heemin32 Ack, removed IT tests but kept unit tests.

Reply:
> What is the case where unit test fails to catch but IT can catch here?

The case where someone wraps another exception after k-NN has thrown the excpetion.


> The responsibility of return code is not in your code but in the OpenSearch framework. All you need to do is throwing correct exception.

Yes responsibility lies with Opensearch to send the correct status code, but we are relying on a behavior of Opensearch and as this exception is thrown from k-NN side we need to make sure that our customers are getting right status code. Otherwise we could have created a new exception of our own.


A simple understanding of adding an IT here is, now on the rest layer a different response will be returned hence we need to make sure that same response code is received by customer what we are expecting it to receive, and in this case that is 400.


Also, having more tests is always better. I am not able to understand why there is harm is having an IT which tests negative scenarios for k-NN.

Reply:
> The case where someone wraps another exception after k-NN has thrown the excpetion.
1. The case is rare. 2. If this is high concern, we need a better mechanism to prevent it rather than relying on integration test.

> Having more tests is always better.
Yes. However, it is not free but it comes with costs: longer test time, engineering efforts on implementation and maintenance.

Reply:
> If this is high concern, we need a better mechanism to prevent it rather than relying on integration test.

as now integration test is the best way. Another thing is the change which is being done in this PR is changing the RestStatus code, so having a IT is must for cases like this.

Reply:
I don't think OS core will do anything to change the behavior on existing exception. If this is concern we need to have our own exception and code can be verified using unit test.

If the concern is someone wrap exception inside knn repo, we can also validate it by writing a unit test on the most outer method.

Reply:
Thanks all of you on the feedbacks. I've kept the integration tests. They offer a broader check against system-wide issues that unit tests might miss. We'll strive for a balance between comprehensive testing and efficiency.